assignment 6/promisesassignment.js

```diff
@@ -0,0 +1,53 @@
+var fs = require("fs");
+
+var firstMethod = function() {
+  var promise = new Promise(function(resolve, reject) {
+    fs.readFile("file1.txt", function(err, data) {
+      if (err) {
+        reject(err);
+      }
+      console.log("first method completed");
+      resolve(data);
+    });
+  });
+  return promise;
+};
+
+var secondMethod = function(data) {
+  var promise = new Promise(function(resolve, reject) {
+    fs.appendFile("file2.txt", data, function(err) {
+      if (err) {
+        reject(err);
+      }
+      resolve(console.log("second method completed"));
+    });
+  });
+  return promise;
+};
+
+var thirdMethod = function() {
+  var promise = new Promise(function(resolve, reject) {
+    fs.readFile("file2.txt", function(err, data) {
+      if (err) {
```

Comment:
@abansal26 Can we make reject error function separate?

Reply:
yes we can but we will have to initialize seperately. For eg.
fs.readFile("file.txt",function(err){......},function(null,data){......})